Autopilot/Test/Src/PathManager/Test_PathManager_Fsm.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+* Author: Dhruv Rawat
+*/
+
+#include "fff.h"
+#include <gtest/gtest.h>
+
+#include "pathManager.hpp"
+#include "pathStateClasses.hpp"
+
+#include "waypointManager.hpp"
+#include "cruisingState.hpp"
+#include "CommWithAttitudeManager.hpp"
+
+#include "SensorFusion.hpp"
+#include "AttitudePathInterface.hpp"
+
+
+using namespace std; 
+using ::testing::Test;
+
+/***********************************************************************************************************************
+ * Test Fixtures
+ **********************************************************************************************************************/
+
+FAKE_VALUE_FUNC(SFOutput_t, SF_GetResult);
+FAKE_VOID_FUNC(AutoSteer_Init);
+FAKE_VOID_FUNC(CommWithAMInit);
+FAKE_VOID_FUNC(SendCommandsForAM, CommandsForAM*);
+FAKE_VALUE_FUNC(bool, GetAttitudeData, AttitudeData*);
+FAKE_VOID_FUNC(AutoSteer_ComputeCoordinatedTurn, CoordinatedTurnInput_t*, CoordinatedTurnAttitudeManagerCommands_t*);
+FAKE_VOID_FUNC(AutoSteer_ComputeAltitudeAndAirspeed, AltitudeAirspeedInput_t*, AltitudeAirspeedCommands_t*);
+
+class PathManagerFSM : public ::testing::Test
+{
+	public:
+
+		virtual void SetUp()
+		{
+			RESET_FAKE(SF_GetResult);
+			RESET_FAKE(AutoSteer_Init);
+			RESET_FAKE(CommWithAMInit);
+			RESET_FAKE(SendCommandsForAM);
+			RESET_FAKE(GetAttitudeData);
+			RESET_FAKE(AutoSteer_ComputeCoordinatedTurn);
+			RESET_FAKE(AutoSteer_ComputeAltitudeAndAirspeed);
+		}
+
+		virtual void TearDown()
+		{
+			FFF_RESET_HISTORY();
+		}
+
+};
+
+class PathManagerDataHandoff : public ::testing::Test
+{
+	public:
+
+		virtual void SetUp()
+		{
+			RESET_FAKE(SF_GetResult);
+			RESET_FAKE(AutoSteer_Init);
+			RESET_FAKE(CommWithAMInit);
+			RESET_FAKE(SendCommandsForAM);
+			RESET_FAKE(GetAttitudeData);
+			RESET_FAKE(AutoSteer_ComputeCoordinatedTurn);
+			RESET_FAKE(AutoSteer_ComputeAltitudeAndAirspeed);
+		}
+
+		virtual void TearDown()
+		{
+			FFF_RESET_HISTORY();
+		}
+};
+
+/***********************************************************************************************************************
+ * State Transition Tests (make sure the correct states are reached given some set of circumstances)
+ **********************************************************************************************************************/
+
+
+TEST (PathManagerFSM, InitialStateIsCommsWithAttitude) {
+    /***********************SETUP***********************/
+
+	pathManager pathMan;
+
+	/********************DEPENDENCIES*******************/
```

Comment:
I don't think we need all the comments of "dependencies", "stepthrough", "asserts", on every test.

Reply:
It breaks up the test into distinct sections and makes it more readable. Doesn't hurt too since the compiler removes them anyway in preprocessing

Reply:
True, but half the sections have no content. Tbh, it doesn't really matter, up to you.